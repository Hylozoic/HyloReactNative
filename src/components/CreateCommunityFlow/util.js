export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'
