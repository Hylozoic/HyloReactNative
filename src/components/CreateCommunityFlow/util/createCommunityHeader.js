import React from 'react'
import {
  Text
} from 'react-native'
import { HeaderBackButton } from 'react-navigation'
import { caribbeanGreen, white, white60onCaribbeanGreen } from 'style/colors'

export default function createCommunityHeader (title, navigation) {
  return {
    title: <Text style={styles.headerTitle}>{title}</Text>,
    color: white,
    fontSize: 10,
    headerStyle: {
      backgroundColor: caribbeanGreen,
      borderBottomWidth: 0,
      shadowRadius: 0,
      shadowOffset: {
        height: 0
      },
      elevation: 0
    },
    headerBackgroundColor: caribbeanGreen,
    headerLeft: <HeaderBackButton onPress={() => navigation.goBack()} tintColor={white60onCaribbeanGreen} />
  }
}

const styles = {
  headerTitle: {
    color: white,
    fontSize: 12,
    fontWeight: 'bold'
  }
}
