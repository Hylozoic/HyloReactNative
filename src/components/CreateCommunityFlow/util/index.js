export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (communityDomain) {
  if (!communityDomain) return
  let formattedDomain = communityDomain.replace(/hylo\.com\/c\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/c/' + formattedDomain
  }
  return formattedDomain
}

export function removeUrlFromDomain (communityDomain) {
  if (!communityDomain) return
  return communityDomain.replace('hylo.com/c/', '')
}
