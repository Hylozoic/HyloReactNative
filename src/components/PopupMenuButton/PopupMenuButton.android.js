import React, { Component } from 'react'
import { View, UIManager, findNodeHandle, TouchableOpacity } from 'react-native'

export default class PopupMenuButton extends Component {
  onError (e) {
    console.error('Error opening popupMenu', e)
  }

  onSelect = (action, index) => {
    if (action === 'itemSelected') {
      this.props.onSelect(index)
    }
  }

  onPress = () => {
    if (this.button) {
      UIManager.showPopupMenu(
        findNodeHandle(this.button),
        this.props.actions,
        this.onError,
        this.onSelect
      )
    }
  }

  render () {
    return (
      <View>
        <TouchableOpacity onPress={this.onPress} ref={this.onRef}>
          { this.props.children }
        </TouchableOpacity>
      </View>
    )
  }

  onRef = button => {
    if (!this.button) {
      this.button = button
    }
  }
}
