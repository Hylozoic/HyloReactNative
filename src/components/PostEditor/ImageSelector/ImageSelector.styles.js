import { rhino10, limedSpruce } from 'style/colors'

export default {
  addImageButton: {
    width: 100,
    height: 100,
    borderRadius: 4,
    backgroundColor: rhino10,
    justifyContent: 'center',
    overflow: 'hidden'
  },
  addImageButtonIcon: {
    color: limedSpruce,
    width: '100%',
    textAlign: 'center'
  },
  addImageButtonImage: {
    width: 100,
    height: 100,
    borderRadius: 4
  },
  hiddenImagePicker: {
    width: 0,
    height: 0
  }
}
