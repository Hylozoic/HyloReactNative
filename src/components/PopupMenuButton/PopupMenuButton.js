import React, { useRef, useState } from 'react'
import { UIManager, findNodeHandle, TouchableOpacity } from 'react-native'
import { isIOS } from 'util/platform'
import useHyloActionSheet from 'hooks/useHyloActionSheet'

export default function PopupMenuButton ({
  actions,
  // If you prefer the PopupMenu Style for Android
  useAndroidPopup,
  children,
  viewProps,
  style,
  hitSlop = { top: 5, bottom: 5, left: 5, right: 5 }
}) {
  const { showHyloActionSheet } = useHyloActionSheet()
  const buttonRef = useRef()
  const [open, setOpen] = useState()

  const showAndroidPopup = () => {
    if (buttonRef.current && !open) {
      UIManager.showPopupMenu(
        findNodeHandle(buttonRef.current),
        actions.map(a => a[0]),
        // onError
        e => {
          setOpen(false)

          throw new Error('Error opening popup')
        },
        // onSelect
        (action, index) => {
          setOpen(false)

          if (action === 'itemSelected') {
            actions[index][1]()
          }
        }
      )
    }
  }

  const handlePress = !isIOS && useAndroidPopup
    ? showAndroidPopup
    : () => showHyloActionSheet({ actions })

  return (
    <TouchableOpacity
      onPress={handlePress}
      style={style}
      hitSlop={hitSlop}
      {...viewProps}
      ref={buttonRef}
    >
      {children}
    </TouchableOpacity>
  )
}
