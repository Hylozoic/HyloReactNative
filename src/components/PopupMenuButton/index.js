import PopupMenuButton from './PopupMenuButton'

export default PopupMenuButton
