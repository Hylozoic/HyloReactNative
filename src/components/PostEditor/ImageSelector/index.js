import React from 'react'
import ImagePicker from '../../ImagePicker'
import PopupMenuButton from '../../PopupMenuButton'
import { Image, View } from 'react-native'
import styles from './ImageSelector.styles'

export default function ImageSelector ({ id, type, style, imageUrls, onAdd, onRemove }) {
  return <View style={[styles.container, style]}>
    {imageUrls.map((url, index) => renderImageButton(url, index, onRemove))}
    <ImagePicker
      style={styles.addImageButton}
      iconStyle={styles.addImageButtonIcon}
      onError={err => console.log(err)}
      onChoice={onAdd}
      type={type}
      id={id} />
  </View>
}
ImageSelector.defaultProps = {imageUrls: []}

function renderImageButton (imageUrl, buttonIndex, onRemove) {
  return <PopupMenuButton
    key={imageUrl}
    style={styles.addImageButton}
    actions={['Remove image']}
    destructiveButtonIndex={0}
    onSelect={menuIndex => onRemove(imageUrl)}>
    <Image style={styles.addImageButtonImage} source={{uri: imageUrl}} />
  </PopupMenuButton>
}
