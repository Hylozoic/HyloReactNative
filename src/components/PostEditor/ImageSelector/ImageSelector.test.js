import React from 'react'
import TestRenderer from 'react-test-renderer'
import ImageSelector from './index'
import { Provider } from 'react-redux'
import { createMockStore } from 'util/testing'
import PopupMenuButton from '../../PopupMenuButton'

jest.mock('util/platform', () => ({isIOS: true}))
jest.mock('../../PopupMenuButton', () => 'PopupMenuButtonMock')

let store

beforeEach(() => {
  store = createMockStore()
})

it('renders with no images', () => {
  const node = TestRenderer.create(<Provider store={store}>
    <ImageSelector />
  </Provider>)
  expect(node).toMatchSnapshot()
})

it('renders with images', () => {
  const node = TestRenderer.create(<Provider store={store}>
    <ImageSelector imageUrls={[
      'http://foo.com/foo.png',
      'http://bar.com/bar.png',
      'http://baz.com/baz.png'
    ]} />
  </Provider>)
  expect(node).toMatchSnapshot()
})

it('has an onRemove prop', () => {
  const onRemove = jest.fn()
  const node = TestRenderer.create(<Provider store={store}>
    <ImageSelector imageUrls={['http://foo.com/foo.png']}
      onRemove={onRemove} />
  </Provider>)

  node.root.findByType(PopupMenuButton).props.actions[0][1]()
  expect(onRemove).toHaveBeenCalledWith('http://foo.com/foo.png')
})
