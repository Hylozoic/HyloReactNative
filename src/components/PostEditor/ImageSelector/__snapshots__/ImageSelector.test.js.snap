// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with images 1`] = `
<View>
  <PopupMenuButtonMock
    actions={
      Array [
        Array [
          "Remove image",
          [Function],
        ],
      ]
    }
    destructiveButtonIndex={0}
  >
    <Image
      source={
        Object {
          "uri": "http://foo.com/foo.png",
        }
      }
    />
  </PopupMenuButtonMock>
  <PopupMenuButtonMock
    actions={
      Array [
        Array [
          "Remove image",
          [Function],
        ],
      ]
    }
    destructiveButtonIndex={0}
  >
    <Image
      source={
        Object {
          "uri": "http://bar.com/bar.png",
        }
      }
    />
  </PopupMenuButtonMock>
  <PopupMenuButtonMock
    actions={
      Array [
        Array [
          "Remove image",
          [Function],
        ],
      ]
    }
    destructiveButtonIndex={0}
  >
    <Image
      source={
        Object {
          "uri": "http://baz.com/baz.png",
        }
      }
    />
  </PopupMenuButtonMock>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Icon
      color={undefined}
      name="AddImage"
    />
  </View>
</View>
`;

exports[`renders with no images 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Icon
      color={undefined}
      name="AddImage"
    />
  </View>
</View>
`;
